code/game/gamemodes/objective.dm

```diff
@@ -174,10 +174,10 @@ var/list/potential_theft_objectives=subtypesof(/datum/theft_objective) \
 	check_completion()
 		if(target && target.current)
 			// TODO: Tie into space manager
-			if(target.current.stat == DEAD || issilicon(target.current) || isbrain(target.current) || target.current.z > ZLEVEL_DERELICT || !target.current.ckey) //Borgs/brains/AIs count as dead for traitor objectives. --NeoFite
+			if(target.current.stat == DEAD || issilicon(target.current) || isbrain(target.current) || !target.current.ckey) //Borgs/brains/AIs count as dead for traitor objectives. --NeoFite
 				return 1
 			// TODO: Tie into space manager
-			if((target.current.z in config.admin_levels))
+			if((target.current.z in config.admin_levels) || ((target.current.loc.type in typesof(/obj/structure/closet)) && (target.current.loc.z in config.admin_levels))) //No hiding in lockers and cheezing greentext.

```

Comment:
just change this to use get_turf instead of an awful snowflake for specifically closets